napari/utils/action_manager.py

```diff
@@ -216,15 +216,20 @@ def bind_button(
         """
         self._validate_action_name(name)
 
-        if (action := self._actions.get(name)) and isgeneratorfunction(action):
+        if (action := self._actions.get(name)) and isgeneratorfunction(
+            action.command
+        ):
```

Comment:
It will work only if the action is defined before the bind button. If it is defined later then the check will not work.